provider/debug_logger.go

```diff
@@ -0,0 +1,21 @@
+package provider
+
+import (
+	"fmt"
+	"log"
+	"strings"
+)
+
+type DebugLogger struct{}
+
+func (d *DebugLogger) Printf(format string, v ...interface{}) {
+	log.Printf("[DEBUG] [rediscloud-go-api] "+format, v...)
```

Comment:
We should consider changing this to the new terraform-plugin-log module if it isn't much work
https://developer.hashicorp.com/terraform/plugin/log/writing

Reply:
This may need changes to https://github.com/RedisLabs/rediscloud-go-api to support specifying a logger with or without a context

Reply:
As per [this advice](https://developer.hashicorp.com/terraform/plugin/log/writing#legacy-logging), I've split into two debugLoggers, one for each provider. Pushing the change now...